pandas/io/rdata.py

```diff
@@ -0,0 +1,598 @@
+import io
+import os
+from tempfile import TemporaryDirectory
+from typing import (
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+from pandas._typing import (
+    Buffer,
+    CompressionOptions,
+    FilePathOrBuffer,
+    StorageOptions,
+)
+from pandas.compat._optional import import_optional_dependency
+from pandas.errors import AbstractMethodError
+from pandas.util._decorators import doc
+
+from pandas.core.dtypes.common import is_list_like
+
+from pandas.core.frame import DataFrame
+from pandas.core.shared_docs import _shared_docs
+
+from pandas.io.common import (
+    file_exists,
+    get_handle,
+    is_fsspec_url,
+    is_url,
+    stringify_path,
+)
+
+
+@doc(storage_options=_shared_docs["storage_options"])
+def read_rdata(
+    path_or_buffer: FilePathOrBuffer,
+    file_format: str = "infer",
+    select_frames: Optional[List[str]] = None,
+    rownames: bool = True,
+    storage_options: StorageOptions = None,
+) -> Union[DataFrame, Dict[str, DataFrame]]:
+    r"""
+    Read R data (.RData, .rda, .rds) into DataFrame or ``dict`` of DataFrames.
+
+    .. versionadded:: 1.3.0
+
+    Parameters
+    ----------
+    path_or_buffer : str, path object, or file-like object
+        Any valid file path is acceptable. The string could be a URL.
+        Valid URL schemes include http, ftp, s3, and file.
+
+    file_format : {{'infer', 'rdata', 'rda', 'rds'}}, default 'infer'
+        R serialization type as output from R's base::save or base::saveRDS
+        commands. Default 'infer' will use extension in file name to
+        to determine the format type.
+
+    select_frames : list, default None
+        Selected names of DataFrames to return from R rda and RData types that
+        can contain multiple objects.
+
+    rownames : bool, default True
+        Include original rownames in R data frames to map into a DataFrame index.
+
+    {storage_options}
+
+    Returns
+    -------
+    DataFrame or dict of DataFrames
+        Depends on R data type where rds formats returns a single DataFrame and
+        rda or RData formats return ``dict`` of DataFrames.
+
+    See Also
+    --------
+    read_sas : Read SAS datasets into DataFrame.
```

Comment:
i think you lited read_parquet / read_feather above (ok to add more, but would add in both places the same list)